### PR TITLE
[Messaging] Bump Microsoft.Azure.Amqp version

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to bump the version of Microsoft.Azure.Amqp to v2.6.6 to include a fix for an internal `NullReferenceException` within the transport.  Also included are some corrections for a bad copy/paste in the `ServiceBusReceiver` doc comments.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 ### Other Changes
 
+- Updated the `Microsoft.Azure.Amqp` dependency to 2.6.6, which includes a bug fix for an internal `NullReferenceException` that would sometimes impact creating new links. _(see: [#258]( Azure/azure-amqp#258))_
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md:13]:
```suggestion
- Updated the `Microsoft.Azure.Amqp` dependency to 2.6.6, which includes a bug fix for an internal `NullReferenceException` that would sometimes impact creating new links. _(see: [#258](https://github.com/azure/azure-amqp/issues/258))_
```

File: sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 ### Other Changes
 
+- Updated the `Microsoft.Azure.Amqp` dependency to 2.6.6, which includes a bug fix for an internal `NullReferenceException` that would sometimes impact creating new links. _(see: [#258]( Azure/azure-amqp#258))_
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md:13]:
```suggestion
- Updated the `Microsoft.Azure.Amqp` dependency to 2.6.6, which includes a bug fix for an internal `NullReferenceException` that would sometimes impact creating new links. _(see: [#258](https://github.com/azure/azure-amqp/issues/258))_
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,10 @@
 
 - Fixed issue where the `SupportOrdering` property was not being respected when set on `CreateTopicOptions`.
 
+### Other Changes
+
+- Updated the `Microsoft.Azure.Amqp` dependency to 2.6.6, which includes a bug fix for an internal `NullReferenceException` that would sometimes impact creating new links. _(see: [#258]( Azure/azure-amqp#258))_
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/CHANGELOG.md:17]:
```suggestion
- Updated the `Microsoft.Azure.Amqp` dependency to 2.6.6, which includes a bug fix for an internal `NullReferenceException` that would sometimes impact creating new links. _(see: [#258](https://github.com/azure/azure-amqp/issues/258))_
```

**Review by christothes — APPROVED** [review]

**Comment by jsquire** [comment]
Test failure is due to an unrelated test hang.  Ignoring and merging.

**Pull request merged**